crates/markdown_preview/src/markdown_parser.rs

```diff
@@ -448,13 +452,20 @@ impl<'a> MarkdownParser<'a> {
                     inside_list_item = true;
 
                     // Check for task list marker (`- [ ]` or `- [x]`)
-                    if let Some(next) = self.current() {
-                        match next.0 {
-                            Event::TaskListMarker(checked) => {
-                                task_item = Some(checked);
-                                self.cursor += 1;
+                    if let Some(event) = self.current_event() {
+                        // If there is a linebreak in between two list items the task list marker will actually be the first element of the paragraph
+                        if event == &Event::Start(Tag::Paragraph) {
+                            self.cursor += 1;
+                        }
+
+                        if let Some(event) = self.current_event() {
```

Comment:
NIT: It looks you could do smth. like `if let Some(Event::TaskListMarker(checked)) = self.current_event()` here and reduce the nested things a bit.